### PR TITLE
[x265] Add HDR and high bit depth option for x265

### LINKED ISSUE #13875: [x265] add multilib support
Add x265 compilation options with multilib support for 8, 10 and 12 bit color depths.

### PR DESCRIPTION
**Pull request by ThisIsAreku**
**Describe the pull request**

- What does your PR fix?

Fixes #13875

- Which triplets are supported/not supported? Have you updated the CI baseline?

No changes in support.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes

### CODE REVIEW

File: ports/x265/portfile.cmake [hunk 1]
```diff
@@ -16,14 +16,28 @@ if (VCPKG_TARGET_IS_WINDOWS)
...
+    hdr10plus   ENABLE_HDR10_PLUS
+)
+
+set(X265_DEPTH "8")
+if ("main10" IN_LIST FEATURES)
+    set(X265_DEPTH "10")
+endif()
+if ("main12" IN_LIST FEATURES)
+    set(X265_DEPTH "12")
+endif()
```
Comment (JackBoosY) [inline ports/x265/portfile.cmake:30]: We need to consider the coexistence of these two features.

File: ports/x265/CONTROL [hunk 2]
```diff
@@ -2,3 +2,18 @@ Source: x265
 Version: 3.4
```
Comment (JackBoosY) [inline ports/x265/CONTROL:2]:
```suggestion
Version: 3.4
Port-Version: 1
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Pinging @ThisIsAreku  for response. Is work still being done for this PR?

File: ports/x265/portfile.cmake (hunk 1 above)
Comment (ThisIsAreku) [reply ports/x265/portfile.cmake]: Actually `main10` and `main12` cannot coexist. When building for high bit depth the default is `main10`. I've removed the `main10` feature.

**Comment by ThisIsAreku** [comment]
@JackBoosY a new commit has been pushed. I'm waiting for the tests to complete

File: ports/x265/CONTROL [hunk 3]
```diff
@@ -1,5 +1,16 @@
 Source: x265
 Version: 3.4
-Port-Version: 2
+Port-Version: 3
 Homepage: https://github.com/videolan/x265
 Description: x265 is a H.265 / HEVC video encoder application library, designed to encode video or images into an H.265 / HEVC encoded bitstream.
+
```
Comment (JackBoosY) [inline ports/x265/CONTROL:6]:
```suggestion
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Needs test features.

**Comment by JackBoosY** [comment]
When building `x265[hbd,main12,hdr10plus]`:
```
G:\14572\vcpkg\buildtrees\x265\src\f1b45711b7-44018d5a0c.clean\source\common\x86\ssd-a.asm:827: error: invalid combination of opcode and operands
```

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**